segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/IndexAppendProcessor.java

```diff
@@ -69,15 +77,42 @@ private void handleIndexAppend(String segmentName) {
              .whenComplete((length, ex) -> {
                  if (ex == null) {
                      log.trace("Index segment append successful for Segment : {} with length {} ", segmentName, length);
+                 } else if (Exceptions.unwrap(ex) instanceof StreamSegmentNotExistsException) {
+                     logWarningMessage(segmentName, ex);
+                     createIndexSegment(segmentName);
                  } else {
-                     log.warn("Index segment append failed for segment {} due to ", segmentName, ex);
+                     logWarningMessage(segmentName, ex);
```

Comment:
I don't think we want these two message to be identical, it makes it hard to tell which path was taken when looking at the log.

Reply:
Done.